distributed_index_tests/lib/user.py

```diff
@@ -1,11 +1,188 @@
-import base64
+import requests
+import concurrent.futures
 import json
+import pprint
+import copy_reg
+import types
+import time
+import base64
+import document
+import uuid
+import re
+from collections import defaultdict
+
+# This function is added to use ProcessExecutor
+# concurrent.futures. 
+# 
+def _pickle_method(m):
+    if m.im_self is None:
+        return getattr, (m.im_class, m.im_func.func_name)
+    else:
+        return getattr, (m.im_self, m.im_func.func_name)
 
-class User:
 
-    def __init__(self, name, password, channels):
+copy_reg.pickle(types.MethodType, _pickle_method)
+
+
+class User:
+    def __init__(self,target_url,db,name,password,channels,uuid=False):
+        self.target_url = target_url
         self.name = name
         self.password = password
-        self.channels = channels
-        auth = base64.b64encode("{0}:{1}".format(self.name, self.password).encode())
-        self.auth = auth.decode("UTF-8")
+        self.db = db
+        self.docs = {}
+        self.use_uuid_doc_names = uuid
```

Comment:
Is this a property that is necessary to the test scenario code? If, not we should find a way to contain it in the class.